app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\CompanyModel;
+use App\Http\Requests\UserRequest;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Hash;
+use App\Utils\StrUtil;
+
+class UserController extends Controller
+{
+    public function index() {
+        $allUsers = User::withTrashed()->paginate(10);
+        return view('user.index', compact('allUsers'));
+    }
+
+
+    public function view(FormRequest $request ,$id) {
+        $user = User::find($id);
+        if (!$user) {
+            return redirect()->back()->with([
+                'type' => 'error',
+                'message' => 'User not found',
+            ]);
+        }
+        $companies = CompanyModel::all();
+        return view('user.view', compact('user', 'companies'));
+    }
+
+    public function store(UserRequest $request, $id) {
+        // $validatedData = $request->validate();
+        $user = User::find($id);
+        if (!empty($request->avatar)) {
+            // dd(1);
+            $user->avatar = uploadFile($request->avatar, 'avatars');
+        }
+        $user->name = $request->name;
+        $user->role = $request->role;
+        $user->company_id = $request->company !== '0' ? $request->company : null;
+        $user->dob = $request->dob;
+        if (!empty($request->password)) {
+            $user->password = Hash::make($request->password);
+        }
+        $user->save();
+        $toast = [
+            'type' => 'success',
+            'message' => 'User updated successfully!'
+        ];
```

Comment:
toàn bộ phần xử lý logic nên đưa vào service riêng rồi gọi ra, sử dụng Dependency Injection